patchwork-fml/src/main/java/net/minecraftforge/fml/client/event/ConfigChangedEvent.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2019, 2019
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.fml.client.event;
+
+import javax.annotation.Nullable;
+
+import net.minecraftforge.eventbus.api.Event;
+
+/**
+ * These events are posted from the GuiConfig screen when the done button is pressed. The events are only posted
+ * if the parent screen is not an instance of GuiConfig or if the configID field has been set for
+ * the GuiConfig screen.
+ *
+ * <p>Listeners for this event should use OnConfigChanged or PostConfigChanged and check for a specific mod ID.
+ * For best results the listener should refresh any objects/fields that are set based on the mod's config
+ * and should serialize the modified config.</p>
+ *
+ * <p>TODO: These events will never be fired by Patchwork (until a ModMenu integration?)</p>
+ * @author bspkrs
+ */
+public class ConfigChangedEvent extends Event {
+	private final String modID;
+	private final boolean isWorldRunning;
+	private final boolean requiresMcRestart;
+	@Nullable
+	private final String configID;
+
+	public ConfigChangedEvent(String modID, @Nullable String configID, boolean isWorldRunning, boolean requiresMcRestart) {
+		this.modID = modID;
+		this.configID = configID;
+		this.isWorldRunning = isWorldRunning;
+		this.requiresMcRestart = requiresMcRestart;
+	}
+
+	@Override
+	public boolean hasResult() {
+		return true;
+	}
+
+	/**
+	 * The Mod ID of the mod whose configuration just changed.
+	 */
+	public String getModID() {
+		return modID;
+	}
+
+	/**
+	 * Whether or not a world is currently running.
+	 */
+	public boolean isWorldRunning() {
+		return isWorldRunning;
+	}
+
+	/**
+	 * Will be set to true if any elements were changed that require a restart of Minecraft.
+	 */
+	public boolean isRequiresMcRestart() {
+		return requiresMcRestart;
+	}
+
+	/**
+	 * A String identifier for this ConfigChangedEvent.
+	 */
+	@Nullable
+	public String getConfigID() {
+		return configID;
+	}
+
+	/**
```

Comment:
I'd like this javadoc to be rewritten to be more like HTML.

* Use `<p>` tags where appropriate
* Use an unordered list (`<ul>`) for the bullet points
* Use an ordered list (`<ol>`) for the 1) and 2)

Also the same criticism about `GuiConfig` applies from the parent class.

Reply:
Since this will have to be re implemented in-house anyways, I'm going to simply the javadoc.

Reply:
:+1: